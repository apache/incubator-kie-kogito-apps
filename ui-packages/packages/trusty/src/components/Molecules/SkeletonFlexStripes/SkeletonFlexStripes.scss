.skeleton-flex-stripes {
  &--padded {
    padding: 0.5em 1em;
  }
}
