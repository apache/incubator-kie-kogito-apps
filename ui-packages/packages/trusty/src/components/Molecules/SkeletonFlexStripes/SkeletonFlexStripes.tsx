/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { Flex, FlexItem } from '@patternfly/react-core';
import SkeletonStripe from '../../Atoms/SkeletonStripe/SkeletonStripe';
import './SkeletonFlexStripes.scss';

type SkeletonFlexStripesProps = {
  stripesNumber: number;
  stripesWidth: string;
  stripesHeight: string;
  isPadded?: boolean;
};

const SkeletonFlexStripes = (props: SkeletonFlexStripesProps) => {
  const { stripesNumber, stripesWidth, stripesHeight, isPadded = true } = props;
  const stripes = [];
  const stripeStyle = {
    width: stripesWidth,
    height: stripesHeight
  };
  const className = isPadded ? 'skeleton skeleton--padded' : 'skeleton';
  for (let i = 0; i < stripesNumber; i++) {
    stripes.push(
      <FlexItem key={`skeleton-${i}`}>
        <SkeletonStripe customStyle={stripeStyle} />
      </FlexItem>
    );
  }
  return <Flex className={className}>{stripes}</Flex>;
};

export default SkeletonFlexStripes;
