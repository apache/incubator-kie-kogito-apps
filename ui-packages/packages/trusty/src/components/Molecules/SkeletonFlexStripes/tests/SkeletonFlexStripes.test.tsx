import React from 'react';
import { shallow } from 'enzyme';
import SkeletonFlexStripes from '../SkeletonFlexStripes';

describe('SkeletonFlexStripes', () => {
  test('renders a list of three stripes', () => {
    const wrapper = shallow(
      <SkeletonFlexStripes
        stripesNumber={3}
        stripesWidth="200px"
        stripesHeight="1em"
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('className')).toMatch('skeleton__flex-stripes');
    expect(wrapper.find('SkeletonStripe')).toHaveLength(3);
    expect(
      wrapper
        .find('SkeletonStripe')
        .at(0)
        .prop('customStyle')
    ).toStrictEqual({
      width: '200px',
      height: '1em'
    });
  });

  test('renders a padded list of stripes', () => {
    const wrapper = shallow(
      <SkeletonFlexStripes
        stripesNumber={2}
        stripesWidth="200px"
        stripesHeight="1em"
        isPadded
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('className')).toMatch(
      'skeleton__flex-stripes skeleton__flex-stripes--padded'
    );
  });
});
