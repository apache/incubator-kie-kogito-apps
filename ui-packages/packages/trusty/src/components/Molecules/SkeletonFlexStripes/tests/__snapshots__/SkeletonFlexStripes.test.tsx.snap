// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonFlexStripes renders a list of three stripes 1`] = `
<Flex
  className="skeleton skeleton--padded"
>
  <FlexItem
    key="skeleton-0"
  >
    <SkeletonStripe
      customStyle={
        Object {
          "height": "1em",
          "width": "200px",
        }
      }
    />
  </FlexItem>
  <FlexItem
    key="skeleton-1"
  >
    <SkeletonStripe
      customStyle={
        Object {
          "height": "1em",
          "width": "200px",
        }
      }
    />
  </FlexItem>
  <FlexItem
    key="skeleton-2"
  >
    <SkeletonStripe
      customStyle={
        Object {
          "height": "1em",
          "width": "200px",
        }
      }
    />
  </FlexItem>
</Flex>
`;

exports[`SkeletonFlexStripes renders a padded list of stripes 1`] = `
<Flex
  className="skeleton skeleton--padded"
>
  <FlexItem
    key="skeleton-0"
  >
    <SkeletonStripe
      customStyle={
        Object {
          "height": "1em",
          "width": "200px",
        }
      }
    />
  </FlexItem>
  <FlexItem
    key="skeleton-1"
  >
    <SkeletonStripe
      customStyle={
        Object {
          "height": "1em",
          "width": "200px",
        }
      }
    />
  </FlexItem>
</Flex>
`;
